static/src/javascripts/projects/common/modules/discussion/comments.js

```diff
@@ -0,0 +1,564 @@
+        }
+
+        let promise;
+        const ajaxParams = { mode: 'cors' };
+
+        if (this.isAllPageSizeActive()) {
+            promise = new WholeDiscussion({
+                discussionId: this.options.discussionId,
+                orderBy: queryParams.orderBy,
+                displayThreaded: queryParams.displayThreaded,
+                maxResponses: queryParams.maxResponses,
+                commentsClosed: queryParams.commentsClosed,
+            })
+                .loadAllComments()
+                .catch(() => {
+                    this.wholeDiscussionErrors = true;
+                    queryParams.pageSize = 100;
+                    return fetchJson(
+                        `${url}?${constructQuery(queryParams)}`,
+                        ajaxParams
+                    );
+                });
+        } else {
+            // It is possible that the user has chosen to view all comments,
+            // but the WholeDiscussion module has failed. Fall back to 100 comments.
+            if (queryParams.pageSize === 'All') {
+                queryParams.pageSize = 100;
+            }
+            promise = fetchJson(
+                `${url}?${constructQuery(queryParams)}`,
+                ajaxParams
+            );
+        }
+
+        return promise.then(this.renderComments.bind(this));
+    }
+
+    unPickComment(commentId: string, $thisButton: bonzo): reqwestPromise {
+        const comment = qwery(`#comment-${commentId}`);
+
+        return DiscussionApi.unPickComment(commentId).then(() => {
+            $(this.getClass('commentPick'), comment).addClass('u-h');
+            $thisButton.text('Pick');
+            comment.setAttribute('data-comment-highlighted', false);
+        });
+    }
+
+    pickComment(commentId: string, $thisButton: bonzo): reqwestPromise {
+        const comment = qwery(`#comment-${commentId}`, this.elem);
+
+        return DiscussionApi.pickComment(commentId).then(() => {
+            $(this.getClass('commentPick'), comment).removeClass('u-h');
+            $thisButton.text('Unpick');
+            comment.setAttribute('data-comment-highlighted', true);
+        });
+    }
+
+    handlePickClick(e: Event) {
+        e.preventDefault();
+
+        const target: HTMLElement = (e.target: any);
+        const commentId = target.getAttribute('data-comment-id');
+        const $thisButton = $(target);
+        const promise =
+            $thisButton[0].getAttribute('data-comment-highlighted') === 'true'
+                ? this.unPickComment
+                : this.pickComment;
+
+        if (commentId) {
+            promise(commentId, $thisButton).fail(resp => {
+                const responseText =
+                    resp.response.length > 0
+                        ? JSON.parse(resp.response).message
+                        : resp.statusText;
+
+                $(target).text(responseText);
+            });
+        }
+    }
+
+    comments: ?qwery;
+    topLevelComments: ?qwery;
```

Comment:
Is it just me, or is it odd that these class properties appear halfway down the class definition?